cmd/templ/lspcmd/proxy/server.go

```diff
@@ -344,21 +346,36 @@ func (p *Server) Completion(ctx context.Context, params *lsp.CompletionParams) (
 			if !ok {
 				continue
 			}
-			imp := addImport(doc.Lines, item.Detail)
-			te := lsp.TextEdit{
-				Range: lsp.Range{
-					Start: lsp.Position{Line: uint32(imp.LineIndex), Character: 0},
-					End:   lsp.Position{Line: uint32(imp.LineIndex), Character: 0},
-				},
-				NewText: imp.Text,
+
+			te := handleImportTextEdit(doc, &item, item.Detail)
+			item.AdditionalTextEdits = []lsp.TextEdit{te}
+		}
+		if strings.Contains(item.Detail, "(from \"") {
+			doc, ok := p.TemplSource.Get(string(templURI))
+			if !ok {
+				continue
 			}
+			pkg := pgkFromImportDetail.FindStringSubmatch(item.Detail)[0]
+			te := handleImportTextEdit(doc, &item, pkg)
```

Comment:
```suggestion
			te := handleImportTextEdit(doc, pkg)
```

item isn't used by handleImportTextEdit.